src/Controller/AbstractBaseCqrsController.php

```diff
@@ -0,0 +1,169 @@
+
+    /* @since [*next-version*] */
+    use CreateInvalidArgumentExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateContainerExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateNotFoundExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use StringTranslatingTrait;
+
+    /**
+     * The bookings resource model.
+     *
+     * @since [*next-version*]
+     *
+     * @var SelectCapableInterface
+     */
+    protected $selectRm;
+
+    /**
+     * The expression builder.
+     *
+     * @since [*next-version*]
+     *
+     * @var object
+     */
+    protected $exprBuilder;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function get($params = [])
+    {
+        $condition = $this->_buildCondition($params);
+        $selected  = $this->selectRm->select($condition);
+
+        $results = [];
+        foreach ($selected as $_idx => $_data) {
+            $results[$_idx] = $this->_createResource($_data);
+        }
+
+        return $results;
+    }
+
+    /**
+     * Builds a logical expression from a set of params.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|ArrayAccess|ContainerInterface $params The input parameters.
+     *
+     * @return LogicalExpressionInterface|null The built condition.
+     */
+    protected function _buildCondition($params)
+    {
+        // The query condition
+        $condition = null;
+
+        foreach ($this->_getParamsInfo() as $_param => $_info) {
+            $_compare = $this->_containerGet($_info, 'compare');
+            $_entity  = $this->_containerGet($_info, 'entity');
+            $_field   = $this->_containerGet($_info, 'field');
+            $_value   = $this->_containerHas($params, $_param)
+                ? $this->_containerGet($params, $_param)
+                : null;
+
+            $condition = $this->_addQueryCondition($condition, $_entity, $_field, $_value, $_compare);
+        }
+
+        return $condition;
+    }
+
+    /**
+     * Adds a query condition to another root condition.
+     *
+     * @since [*next-version*]
+     *
+     * @param LogicalExpressionInterface|null $condition The root condition.
+     * @param string                          $entity    The entity name to use in the expression.
+     * @param string                          $field     The name of field.
+     * @param mixed                           $value     The query value.
+     * @param string                          $compare   The comparison mode - the  expression builder method.
+     *
+     * @return LogicalExpressionInterface|null The amended condition.
+     */
+    protected function _addQueryCondition($condition, $entity, $field, $value, $compare = 'eq')
+    {
+        if ($value === null) {
+            return $condition;
+        }
+
+        $b = $this->exprBuilder;
+
+        $queryCondition = $b->$compare(
+            $b->ef($entity, $field),
```

Comment:
Ideally, running builder methods based on a comparison type code should happen via a mapping, instead of variable method call.